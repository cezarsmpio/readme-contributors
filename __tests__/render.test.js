const {
    render,
    renderContributor,
    renderContributors,
    renderTemplate,
} = require('../render');

jest.mock('../themes.js', function() {
    return {
        getTheme() {
            return 'body { font-size: 16px; }';
        },
    };
});

const contributors = [
    {
        avatar_url: 'https://avatars3.githubusercontent.com/u/0?v=4',
    },
    {
        avatar_url: 'https://avatars3.githubusercontent.com/u/0?v=4',
    },
];

it('should render a contributor item', () => {
    expect(
        renderContributor(contributors[0], { avatarSize: 30 }),
    ).toMatchSnapshot();
});

it('should render a list of contributors', () => {
    expect(
        renderContributors(contributors, { avatarSize: 30 }),
    ).toMatchSnapshot();
});

it('should render', () => {
    expect(
        render(contributors, { theme: 'simple', avatarSize: 25 }),
    ).toMatchSnapshot();
});

it('should render a template', () => {
    expect(
        renderTemplate('Hello, {{ greeting }}!', { greeting: 'world' }),
    ).toBe('Hello, world!');
});

it('should not replace the content of a template', () => {
    expect(renderTemplate('Hey!', { hello: 'world' })).toBe('Hey!');
    expect(renderTemplate('{ hello }', { hello: 'world' })).toBe('{ hello }');
});
