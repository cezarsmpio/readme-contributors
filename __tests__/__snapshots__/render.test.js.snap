// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `"<style>body { font-size: 16px; }</style><ul class=\\"contributors-list\\"><li class=\\"contributor\\"><img src=https://avatars3.githubusercontent.com/u/0?v=4 width=\\"25\\" /></li><li class=\\"contributor\\"><img src=https://avatars3.githubusercontent.com/u/0?v=4 width=\\"25\\" /></li></ul>"`;

exports[`should render a contributor item 1`] = `"<li class=\\"contributor\\"><img src=https://avatars3.githubusercontent.com/u/0?v=4 width=\\"30\\" /></li>"`;

exports[`should render a list of contributors 1`] = `"<ul class=\\"contributors-list\\"><li class=\\"contributor\\"><img src=https://avatars3.githubusercontent.com/u/0?v=4 width=\\"30\\" /></li><li class=\\"contributor\\"><img src=https://avatars3.githubusercontent.com/u/0?v=4 width=\\"30\\" /></li></ul>"`;
